src/components/version-language-switcher/VersionLanguageSwitcher.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  GoabIcon,
+  GoabPopover,
+} from "@abgov/react-components";
+import {
+  ANGULAR_VERSIONS, getVersionedUrlPath, Language, LanguageVersion,
+  OLD_ANGULAR_URL_SEGMENT,
+  OLD_REACT_URL_SEGMENT, REACT_VERSIONS
+} from "./version-language-constants.ts";
+import { useLocation, useNavigate } from "react-router-dom";
+import React, { useContext } from "react";
+import { LanguageVersionContext } from "@contexts/LanguageVersionContext.tsx";
+import "./version-language-switcher.css";
+
+export const VersionLanguageSwitcher = () => {
+  const {language, version, setLanguage, setVersion} = useContext(LanguageVersionContext);
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const updateURL = (key: "language" | "version", newValue: LanguageVersion | Language) => {
+    const isComponentRoute =
+      location.pathname.startsWith("/components/") && location.pathname.split("/").length > 2;
+    let newLanguageValue = language; // fallback value from localStorage
+    let newVersionValue = version; // fallback value from localStorage
+
+    if (key === "language") newLanguageValue = newValue as Language;
+    if (key === "version") newVersionValue = newValue as LanguageVersion;
+
+    const combineSegment = getVersionedUrlPath(newVersionValue, newLanguageValue);
+    if (isComponentRoute) {
+      const pathSegments = location.pathname.split("/");
+      const componentName = pathSegments[pathSegments.length -1];
+      if (OLD_REACT_URL_SEGMENT === combineSegment || OLD_ANGULAR_URL_SEGMENT === combineSegment) {
+        const newPath = `/components/${combineSegment}/${componentName}${location.hash}`;
+        navigate(newPath, {replace: true});
+      } else {
+        const newPath = `/components/${componentName}${location.hash}`;
+        navigate(newPath, {replace: true});
+      }
+    }
+  }
+
+  const updateLanguage = (newValue: "react" | "angular") => {
+    setLanguage(newValue);
+    updateURL("language", newValue);
+  }
+
+  const updateVersion = (newValue: "old" | "new") => {
+    setVersion(newValue);
+    updateURL("version", newValue);
+  }
+
+  const capitalizeFirstLetter = (str: string): string => {
+    if (!str) return str;
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  }
+
+  const openLanguagePopOver = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+  };
+
+  const openVersionPopOver = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+  };
+
+  const getCurrentVersionLabel = (language: string, version: string) => {
+    if (language === "react") {
+      return version === "new" ? REACT_VERSIONS.NEW.label : REACT_VERSIONS.OLD.label;
+    }    if (language === "angular") {
+      return version === "new" ? ANGULAR_VERSIONS.NEW.label : ANGULAR_VERSIONS.OLD.label;
+    }
+  }
+
+  return (
+    <>
+      <GoabPopover
+        target={
```

Comment:
To manually control the popover open state:
```suggestion
        open={isLanguagePopoverOpen}
```
- Add to `openLanguagePopover` handler: `setIsLanguagePopoverOpen(true)`
- Add to `updateLanguage` handler: `setIsLanguagePopoverOpen(false)`

Other Notes:
- [This fix](https://github.com/GovAlta/ui-components/pull/2344) won't help here with controlling popover open state.
- AppHeaderMenu has its own logic to control popover open state, closing popover whenever the url changes. In theory it would be better to use AppHeaderMenu and AppHeaderMenuItem here.